.github/workflows/beta-release.yaml

```diff
@@ -7,85 +7,65 @@ on:
-    permissions:
-      id-token: write
-      contents: read
-    steps:
-      - name: Configure AWS Credentials (build)
-        uses: aws-actions/configure-aws-credentials@v1
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1-node16
         with:
           aws-region: us-west-2
           role-to-assume: ${{ secrets.CI_AWS_ROLE }}
-          role-session-name: ControllerBetaRelease
-      - name: Pull docker image
-        run: |
-          aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin ${{ secrets.CI_AWS_ACCOUNT }}.dkr.ecr.us-west-2.amazonaws.com
-          docker pull ${{ secrets.CI_AWS_ACCOUNT }}.dkr.ecr.us-west-2.amazonaws.com/amazon/appmesh-controller:${{ github.event.inputs.tag }}
-          docker pull ${{ secrets.CI_AWS_ACCOUNT }}.dkr.ecr.us-west-2.amazonaws.com/amazon/appmesh-controller:${{ github.event.inputs.tag }}-linux_arm64
-      - name: Configure AWS Credentials (beta)
-        uses: aws-actions/configure-aws-credentials@v1
-        with:
-          aws-region: us-west-2
-          role-to-assume: ${{ secrets.BETA_AWS_ROLE }}
-          role-session-name: ControllerBetaRelease
-      - name: Push docker image
+          role-session-name: ImagePusher
+
+      - name: Build Images
         run: |
-          aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin ${{ secrets.BETA_AWS_ACCOUNT }}.dkr.ecr.us-west-2.amazonaws.com
-          docker tag ${{ secrets.CI_AWS_ACCOUNT }}.dkr.ecr.us-west-2.amazonaws.com/amazon/appmesh-controller:${{ github.event.inputs.tag }} ${{ secrets.BETA_AWS_ACCOUNT }}.dkr.ecr.us-west-2.amazonaws.com/amazon/appmesh-controller:${{ github.event.inputs.tag }}
-          docker tag ${{ secrets.CI_AWS_ACCOUNT }}.dkr.ecr.us-west-2.amazonaws.com/amazon/appmesh-controller:${{ github.event.inputs.tag }}-linux_arm64 ${{ secrets.BETA_AWS_ACCOUNT }}.dkr.ecr.us-west-2.amazonaws.com/amazon/appmesh-controller:${{ github.event.inputs.tag }}-linux_arm64
-          docker push ${{ secrets.BETA_AWS_ACCOUNT }}.dkr.ecr.us-west-2.amazonaws.com/amazon/appmesh-controller:${{ github.event.inputs.tag }}
-          docker push ${{ secrets.BETA_AWS_ACCOUNT }}.dkr.ecr.us-west-2.amazonaws.com/amazon/appmesh-controller:${{ github.event.inputs.tag }}-linux_arm64
+          aws ecr get-login-password --region us-west-2 | \
+            docker login --username AWS --password-stdin $IMAGE_HOST
+          # Note: right now, this pushes the amd image under the default. This
+          # behavior should be changed to supporting multiarch shortly.
+          docker buildx build \
+            --platform linux/amd64 \
+            --build-arg GOPROXY="$GOPROXY" \
+            -t "${IMAGE}:${IMAGE_TAG}" \
+            . --push
+          docker buildx build \
+            --platform linux/arm64 \
+            --build-arg GOPROXY="$GOPROXY" \
+            -t "${IMAGE}:${IMAGE_TAG_ARM}" \
+            . --push
```

Comment:
I had some earlier logic that did a proper multiarch deploy, but took it out to avoid the already rather broad scope. I can dial back the logic even further if desired to make this look more like the original action.